### DISCUSSION TITLE
Dynamic Form Field update

### DISCUSSION THREAD

**Discussion by embersaw** [react-hook-form/react-hook-form#9577] [Q&A] [answered]
Hi, how can i update a specific field of a dynamic form? For example: 
In here,
![image](https://user-images.githubusercontent.com/63653804/207238391-6c361283-1361-45fe-80ff-9a27b23ea2d4.png)

If I select equipment type, I want to change the capacity. How to accomplish this?

**Comment by bluebill1049** [comment]
```setValue(`fieldName.${index}.name`)``` can you do this?

**Reply by embersaw** [reply]
No, it doesn't work. This is my code snippet.
![image](https://user-images.githubusercontent.com/63653804/207253254-1560e501-4e2d-491e-b7d7-523d7b0c859a.png)

**Comment by Moshyfawn** [comment, marked as answer]
You dependant field update shouldn't be an effect but rather a call on the change event. Something like
```jsx
const { update } = useFieldArray()

return (
  <Controller
    render={({ field }) => (
      <select
        {...field}
        onChange={(event) => {
          // INFO: update dependant fields here 
          update(index, {
            field: event.target.value,
            dep: event.target.value
          });
        }}
      >
        ...
    )}
  />
)
```

**Reply by Moshyfawn** [reply]
Here's a working example: [sandbox](https://codesandbox.io/s/sleepy-oskar-5byy1e?file=%2Fsrc%2FApp.tsx)

**Reply by embersaw** [reply]
Thank you
